packages/runtime/client/rest/src/contract/endpoints/collection.ts

```diff
@@ -211,52 +236,217 @@ export interface CreateContractRequest {
+   *   - Bytestrings are hex-encoded, with a maximum length of 64 bytes.
+   *
+   * Metadata aren't stored as JSON on the Cardano blockchain but are instead stored using a compact binary encoding (CBOR).
+   * The binary encoding of metadata values supports three simple types:
+   *    - Integers in the range `-(2^64 - 1)` to `2^64 - 1`
+   *    - Strings (UTF-8 encoded)
+   *    - Bytestrings
+   *    - And two compound types:
+   *        - Lists of metadata values
+   *        - Mappings from metadata values to metadata values
+   * </p>
+   * It is possible to transform any JSON object into this schema (See https://developers.cardano.org/docs/transaction-metadata )
+   * @see
+   * https://developers.cardano.org/docs/transaction-metadata
    */
-  // TODO: Add link to example of metadata
   metadata?: Metadata;
   /**
-   * To avoid spamming the network, the cardano ledger requires us to deposit a minimum amount of ADA.
-   * The value is in lovelace, so if you want to deposit 3Ada you need to pass 3_000_000 here.
+   * Minimum Lovelace value to add on the UTxO created (Representing the Marlowe Contract Created on the ledger).This value
+   * is computed automatically within the Runtime, so this parameter is only necessary if you need some custom adjustment.
+   *
+   * <h4>Justification</h4>
+   * <p>Creating a Marlowe Contracts over Cardano is about creating UTxO entries on the Ledger.
+   *
+   * To protect the ledger from growing beyond a certain size that will cost too much to maintain,
+   * a constraint called "Minimum ada value requirement (mininmumLovelaceUTxODeposit)" that adjust
+   * the value (in ADA) of each UTxO has been added.
+   *
+   * The more the UTxOs entries are big in size, the more the value of minimum
+   * of ADAs needs to be contained.</p>
+   * @see
+   * https://docs.cardano.org/native-tokens/minimum-ada-value-requirement
    */
-  // TODO: @sam
-  //       Create a global documentation page (and link from here) that explains the concept of minUTxO,
-  //       why it is required, who deposits it, and how and when you get it back.
-  minUTxODeposit: number;
+  mininmumLovelaceUTxODeposit?: number;
 
-  // TODO: Comment this and improve the generated type (currently `string | {}`)
-  roles?: RolesConfig;
   /**
-   * An optional object of tags where the **key** is the tag name (`string`) and the **value** is the tag content (`any`)
+   * @experimental
+   * Thread Roles are a details of implementation within the runtime. It allows provide a custom name
+   * if the thread role name is conflicting with other role names used.
+   * @default
+   *  - the Thread Role Name is "" by default.
    */
-  tags?: Tags;
+  threadRoleName?: RoleName;
+
+  /**
+   * Role Token Configuration for the contract passed in the `contractOrSourceId` field.
+   *
+   * <h4>Prerequisite</h4>
+   * <p>
+   * Participants ({@link @marlowe.io/language-core-v1!index.Party | Party}) in a Marlowe Contract can be expressed in 2 ways:
```

Comment:
More than Prerequisite

<h4>Participants</h4>
A Marlowe contract can have two types of {@link Participants}:
1. Address: When an address is fixed in the contract we don't need to provide further configuration
2. Role: When the participation is done through a Role Token, we need to define if that token is minted as part of the contract creation transaction or if it was minted beforehand.